test/FastTests/Corax/StreamingOptimization.cs

```diff
@@ -0,0 +1,224 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Corax;
+using Corax.Mappings;
+using Corax.Queries;
+using Corax.Queries.SortingMatches;
+using FastTests.Voron;
+using Newtonsoft.Json;
+using Raven.Client.Documents;
+using Raven.Client.Documents.Indexes;
+using Raven.Client.Documents.Queries;
+using Raven.Client.Documents.Session;
+using Raven.Client.Json.Serialization.NewtonsoftJson.Internal;
+using Raven.Server.Documents.Indexes.Persistence.Corax;
+using Raven.Server.Documents.Queries;
+using Sparrow.Json;
+using Sparrow.Server;
+using Sparrow.Threading;
+using Tests.Infrastructure;
+using Xunit;
+using Xunit.Abstractions;
+using Index = Raven.Server.Documents.Indexes.Index;
+
+namespace FastTests.Corax;
+
+public class StreamingOptimization(ITestOutputHelper output) : RavenTestBase(output)
+{
+    [Fact] // TermMatch and asc order on same field with same type => can optimize
+    public async Task SortingMatchIsSkippedOnSingleTermMatch() => await TestQueryBuilder<TermMatch>(session => 
```

Comment:
* StartsWith
* Range


* `from Users where startsWith(Name, 'a') order by Name desc`
* `from Users where Date < $now order by Date desc`

Reply:
`from Users where Date > $now order by Date desc`
`from Users where Date < $now order by Date desc`
`from Users where Date < $now order by Date asc`
`from Users where Date > $now order by Date asc`

Reply:
Compounding:

```
from Posts
where BlogId = $blogId
order by PostedAt desc
limit 5
```

And:

```
from Posts
where BlogId = $blogId and PostedAt < $now
order by PostedAt desc
limit 5
```

Reply:
Index will output `BlogId + RecordSeparator + PostedAt`

Reply:
`Compound.Add([BlogId, PostedAt])`